Fix COOLROOM bug in thermal networks

martin-mosteiro opened the pull request:
As described in #2525, the way `calc_DC_supply` was written created an error in buildings with COOLROOMs if there were also SERVERROOMs present in the case study.


@JIMENOFONSECA perhaps you have some thoughts on the change made on `calc_DC_supply`?

JIMENOFONSECA commented:
@martin-mosteiro i made a change so it works for all cases. if it is ok for you, then go ahead an merge. It works for me

martin-mosteiro commented:
@JIMENOFONSECA Thanks for generalizing this. However, I don't think the code did the same thing as before. Since 0 values are used to indicate the temperatures of systems which are not operated, the current version of the script will give 1E-6 any time one of the systems is not operated – even though it should give the supply temperature of the systems that _are_ operated.

So I'm reverting the changes and refactoring the code to generalize it as you did but while keeping `t0 = t1 = 0.0` as a special case.

Ideally, the temperatures of the systems that are not operated should not be 0... but I that would require time to refactor the code, which we don't have at the moment.

martin-mosteiro commented:
Okay, I have now updated the following:
1. fixed `calc_DC_supply` following @JIMENOFONSECA's solution, but removing a bug by which supply temperatures would've always been 1E-6 or lower (solves #2525).
2. fixed another temperature bug (#2529) by which disconnected buildings were also used to set the operating temperature of the thermal network
3. fixed `COOLROOM` process schedules - I just saw that these would've led to 0 kWh refrigeration for all cases!
